### ISSUE TITLE
ISSUE

### ISSUE THREAD

**Issue by aaronleary** [ch4413/br_code#5]
I am now working through the tool to produce a report for Droitwich Spa High School. I will send the relevant data set in a separate email (Alex has the relevant file). I read it that the mean difference between the pre and the post test is -14.6500488599349. Are you able to verify this to be true?

**Comment by ch4413** [comment]
I will assign this to Alex. He's going to check it to make sure it's behaving as expected.

**Comment by ch4413** [comment]
@tapnshred the mean difference will now be positive as I've switched pre and post test around.

**Comment by aaronleary** [comment]
Hi all,

Please see this school. I have two sets of data for PP. Have I queried the data correctly? 

![pp](https://user-images.githubusercontent.com/29143837/27036177-e7ab14f4-4f7b-11e7-970d-b99f196a1bbe.png)

**Comment by ch4413** [comment]
Hi @aaronleary,

When you say "two sets of data for PP", are you saying two sets: the Yes set and the No set? If so, you can only 'group' the data (not filter the data) on this tab.
Your query above is find the means for data for each unique combination of School2, PP, Type. Your table is showing that for STWCC there are six combinations.

What are you wanting to do? Once I know that, I can see if it's possible at the moment.

Chris

**Comment by aaronleary** [comment]
Hi Chris, 

Ok, I get that, thanks. I just wanted to see the comparison between the Pupil Premium students pre and post tests, and then non PP students pre and post tests. One query though - the data suggests that 1031 non PP students took their pre test but 1062 took their post test? They're meant to be matched pairs?

Thanks,
Aaron

**Comment by ch4413** [comment]
There are two ways of looking at the data:

1) See all data and find statistics for all values: Data Viewer, Mean Score Report
2) Remove unpaired and apply statistics to paired data only: t Test, Progress Report

The Mean Scores are found for unfiltered/unpaired data.  At the moment the app only filters the paired data where paired data is needed for the comparative statistics. If you would like the Mean Scores to only be found for the paired data, I can change this.

**Comment by aaronleary** [comment]
Yes we want the mean scores to only be found for the paired data, please.

**Comment by ch4413** [comment]
Okay, mean scores all paired now. You'll see that post and pre test means both have the same amount students.
To see this - the mean difference given by the t-test tab will be the same as the difference between the means of pre and post tests (I've checked this).

**Comment by ch4413** [comment]
@tapnshred could you review this.

**Comment by aaronleary** [comment]
Just trying to have a play with this. Getting this error:
![image](https://user-images.githubusercontent.com/29143837/27348003-7e770d1a-55ea-11e7-92ae-e781301d5f21.png)

**Comment by ch4413** [comment]
Could you send me the column headings on the file that you are loading into the app? I'm loading in a file with these headings and it is working for me:

post_title;"display_name";"id";"user_id";"post_id";"post_type";"timing";"course_id";"related_id";"points";"percentage";"timespent";"status";"meta_value";"name";"post_title";"acc_ethnicity";"acc_pp";"acc_sen";"acc_misc";"acc_gender";"timing";"school";"class";"school2";"class2"

It looks to me as though a new column has been added to the data or a heading has changed name.

**Comment by aaronleary** [comment]
Sorry - just realised I loaded the wrong file! I loaded up one of the reports produced by Alex's app - not a fresh SQL export. Apologies. Your question has jogged my memory. Thanks.

**Comment by aaronleary** [comment]
Correction. I did upload the correct file. These are the headings:
post_title;"display_name";"id";"user_id";"post_id";"post_type";"timing";"course_id";"related_id";"points";"percentage";"timespent";"status";"corrupt_quiz";"meta_value";"name";"post_title";"acc_ethnicity";"acc_pp";"acc_sen";"acc_misc";"acc_gender";"timing";"

**Comment by ch4413** [comment]
Could you try loading the same file again?

As far as I can see, you've added a new column to the data called "corrupt_quiz". Is this true? I don't have this in my dataset.

I code is a bit messy for this at the moment. I will try to change it.
